orchagent/switchorch.cpp

```diff
@@ -900,6 +900,235 @@ void SwitchOrch::doCfgSwitchHashTableTask(Consumer &consumer)
     }
 }
 
+bool SwitchOrch::setSwitchTrimmingSizeSai(const SwitchTrimming &trim) const
+{
+    sai_attribute_t attr;
+
+    attr.id = SAI_SWITCH_ATTR_PACKET_TRIM_SIZE;
+    attr.value.u32 = trim.size.value;
+
+    auto status = sai_switch_api->set_switch_attribute(gSwitchId, &attr);
```

Comment:
Does vendor SAI return error if the trim size is less than the supported size?

Reply:
@kperumalbfn SAI headers API provides a wide range of trimming size configuration, but in case some values are not acceptable or any other limitations, the vendor SAI will return an error which will be processed by OA